File: src/main/java/wordseeker/gui/table/Table.java

package wordseeker.gui.table;

import java.awt.Font;
import javax.swing.JTable;
import javax.swing.ListSelectionModel;

/**
 * @author rad1kal
 */
public class Table extends JTable {

    public Table() {
        super(TableModel.getInstance());
        setFont(new Font("Verdana", Font.PLAIN, 12));
        setDragEnabled(false);
        setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
        addMouseListener(new TableListener(this));
        setAutoCreateRowSorter(true);
    }

}

File: src/main/java/wordseeker/gui/table/Table.kt
package wordseeker.gui.table

import java.awt.Font
import javax.swing.JTable
import javax.swing.ListSelectionModel

/**
 * @author rad1kal
 */
class Table : JTable(TableModel.instance) {

    init {
        font = Font("Verdana", Font.PLAIN, 12)
        dragEnabled = false
        setSelectionMode(ListSelectionModel.SINGLE_SELECTION)
        addMouseListener(TableListener(this))
        autoCreateRowSorter = true
    }

}
